File: vuln-sandbox/go/cwe-78/command-injection.go
// VULNERABLE: go-exec-command-bash-c — exec.Command with bash -c and user input
// Rule: go-exec-command-bash-c | CWE-78 | Severity: CRITICAL

package main

import (
	"fmt"
	"net/http"
	"os/exec"
)

func runHandler(w http.ResponseWriter, r *http.Request) {
	userCommand := r.URL.Query().Get("cmd")

	// VULNERABLE: user input passed to bash -c allows arbitrary command execution
	// An attacker can pass: ls; cat /etc/passwd to read sensitive files
	cmd := exec.Command("bash", "-c", userCommand)
	out, err := cmd.Output()
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
	fmt.Fprintf(w, "%s", out)
}

func main() {
	http.HandleFunc("/run", runHandler)
	http.ListenAndServe(":8080", nil)
}

File: vuln-sandbox/rust/cwe-78/command_injection.rs
// VULNERABLE: rust-command-shell-sh — Command::new("sh") with -c and user input
// Rule: rust-command-shell-sh | CWE-78 | Severity: CRITICAL

use std::process::Command;

fn run_user_command(user_input: &str) -> String {
    // VULNERABLE: user input passed to sh -c allows arbitrary command execution
    // An attacker can pass: ls; cat /etc/passwd to read sensitive files
    let output = Command::new("sh")
        .arg("-c")
        .arg(user_input)
        .output()
        .expect("failed to execute process");

    String::from_utf8_lossy(&output.stdout).to_string()
}

fn main() {
    // Simulating user input from an HTTP request parameter
    let user_input = std::env::args().nth(1).unwrap_or_default();
    let result = run_user_command(&user_input);
    println!("{}", result);
}
